wallabagger/js/options.js

```diff
@@ -278,83 +303,85 @@ OptionsController.prototype = {
                     this.wallabagUrlNotChecked();
                 });
 
         }
 
     },
 
-    init: function () {
-
-        this.api = new WallabagApi();
-
-        this.api.load().then(data => {
-
-            let wburl = data.Url;
-            let re = /^(http|https):\/\/(.*)/;
-            if (re.test(wburl)) {
-                res = re.exec(wburl);
-                this.protocolCheck_.checked = (res[1] == "https");
-                this.protocolLabel_.innerText = res[1] + "://";
-                this.wallabagurlinput_.value = res[2];
-            };
-            let apiv = data.ApiVersion;
-            if ((apiv != '') && (apiv != null)) {
-                this.versionLabel_.innerHTML = apiv;
-                if (apiv.split('.')[0] == '2') {
-                    this.checkedLabel_.innerHTML = 'OK';
-                    this._green(this.wallabagurlinput_);
-                    this._show(this.tokenSection_);
-                }
-            }
-            let clid = data.ClientId;
-            if ((clid != '') && (clid != null)) {
-                this.clientId_.value = clid;
-            }
-            let clsc = data.ClientSecret;
-            if ((clsc != '') && (clsc != null)) {
-                this.clientSecret_.value = clsc;
-            }
-            let usrlg = data.UserLogin;
-            if ((usrlg != '') && (usrlg != null)) {
-                this.userLogin_.value = usrlg;
-            }
-            let usrp = data.UserPassword;
-            if ((usrp != '') && (usrp != null)) {
-                this.userPassword_.value = usrp;
-            }
-            let atoken = data.ApiToken;
-            if ((atoken != '') && (atoken != null)) {
-                //                this.appTokenInput_.value = atoken;
-                this.tokenLabel_.innerHTML = "Granted";
+    setFields: function (data) {
+
+        console.log('setFields');
+        let wburl = data.Url;
+        let re = /^(http|https):\/\/(.*)/;
+        if (re.test(wburl)) {
+            res = re.exec(wburl);
+            this.protocolCheck_.checked = (res[1] == "https");
+            this.protocolLabel_.innerText = res[1] + "://";
+            this.wallabagurlinput_.value = res[2];
+        };
+        if (this.wallabagurlinput_.value != '') {
+            this._show(this.tokenSection_);
+        }
+        let apiv = data.ApiVersion;
+        if ((apiv != '') && (apiv != null)) {
+            this.versionLabel_.innerHTML = apiv;
+            if (apiv.split('.')[0] == '2') {
+                this.checkedLabel_.innerHTML = 'OK';
+                this._green(this.wallabagurlinput_);
+                this._show(this.tokenSection_);
             }
-            let rtoken = data.RefreshToken;
-            if ((rtoken != '') && (rtoken != null)) {
-                //             this.refrTokenInput_.value = rtoken;
+        }
+        let clid = data.ClientId;
+        if ((clid != '') && (clid != null)) {
```

Comment:
I saw you got the habit to test for `!= '' && != null` but in this case, I don't think this value could ever be null. Undefined at worse.